### ISSUE TITLE
Comparing the RTX 4090 and the RTX 6000 Ada

### ISSUE THREAD

**Issue by ezherman** [nanoporetech/dorado#1181]
I am considering my options for GPUs for live SUP basecalling with a P2 Solo. I get the impression that besides the high-end A100 cards, the RTX 4090 and the RTX 6000 Ada are used with the P2 Solo. Is there any data to inform whether the 6000 cards are worth the price increase compared to the 4090 cards?

**Comment by Kirk3gaard** [comment]
We have the RTX4090s and have been very happy with those. But I agree it would be nice to have some official numbers for benchmarks of basecalling on different GPUs #306 

2x4090s should keep up with SUP on a P2S in realtime running at full speed.

I suspect that we will soon see some 5090s being used for basecalling as well.

**Comment by Psy-Fer** [comment]
Yep, everyone is waiting for Chinese new year to be over so production gets turned back on and we get more stock in Australia. I'll be buying some 5090s when I can.

Also we have a fair amount of benchmarking on a lot of different cards (including AMD). Getting closer to sharing that with everyone.

**Comment by Kirk3gaard** [comment]
SUP v5.0.0 calling with dorado 0.9.1:

![Image](https://github.com/user-attachments/assets/7d318eb4-2c68-4007-94f4-e38550937a86)

**Comment by Psy-Fer** [comment]
Any chance you can put it in Megasamples/s?
This is output by dorado in it's samples/s metric. It's what we have done all our plots in.

We have A2, V100, L40, H100, 3070, 3090 asd well as the ones you have. And hopefully 5090 soon.

Looks like your results are congruent with ours, so that's nice :)

**Comment by Kirk3gaard** [comment]
![Image](https://github.com/user-attachments/assets/e24a900b-6f1e-4fa4-bccd-c8a4e39137b4)
<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/UR36RV/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/UR36RV/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>

</style>
</head>

<body link="#0563C1" vlink="#954F72">


GPU | Speed (samples/s) | CUDA cores | Tensor cores | RAM (GB)
-- | -- | -- | -- | --
L4 | 2,54E+06 | 7424 | 240 | 24
A10 | 3,81E+06 | 9216 | 288 | 24
A100 | 1,11E+07 | 6912 | 432 | 40
A40 | 5,82E+06 | 10752 | 336 | 48
L40S | 8,79E+06 | 18176 | 588 | 48
RTX4090 | 1,06E+07 | 16384 | 512 | 24



</body>

</html>

**Comment by Psy-Fer** [comment]
I'd hate to see the $/samples/s of this table.....

**Comment by Kirk3gaard** [comment]
Random pricing I found on google so there might be quite some variation
![Image](https://github.com/user-attachments/assets/5b884079-8f7f-4bee-a2e8-193019543396)

<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/UR36RV/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/UR36RV/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>

</style>
</head>

<body link="#0563C1" vlink="#954F72">


GPU | Speed (samples/s) | CUDA cores | Tensor cores | RAM (GB) | USD | USD/samples/s
-- | -- | -- | -- | -- | -- | --
L4 | 2,54E+06 | 7424 | 240 | 24 | 3920 | 0,001541
A10 | 3,81E+06 | 9216 | 288 | 24 | 5040 | 0,001321
A100 | 1,11E+07 | 6912 | 432 | 40 | 18200 | 0,001644
A40 | 5,82E+06 | 10752 | 336 | 48 | 6440 | 0,001106
L40S | 8,79E+06 | 18176 | 588 | 48 | 9100 | 0,001035
RTX4090 | 1,06E+07 | 16384 | 512 | 24 | 2520 | 0,000237



</body>

</html>

**Comment by Kirk3gaard** [comment]
Someone should take the next generation of https://tinygrad.org/#tinybox for a spin ;)

**Comment by Psy-Fer** [comment]
4090s in Aus were around $3400AUD, so around $2100 USD. The value proposition of a 4090 is insane. It's why I build PCs with them to run the P2s.

I think to answer the original question asked by @ezherman, an RTX A6000 is not worth the price compared to an RTX 4090, if you can find a 4090.

**Comment by ezherman** [comment]
Incredible, thank you @Kirk3gaard and @Psy-Fer! We are hoping to get our hands on 5090s soon, I'd be happy to run benchmarking on those if there's a protocol that can be shared.

**Comment by Kirk3gaard** [comment]
Have just used dorado as it is with the same [small dataset (kit14,5khz) ](https://www.dropbox.com/scl/fo/zwp8c7400f7p7d1vg8ybc/ADpR4EHHfmWsAlET8eUX4fY?rlkey=trvog3bsv19jy0yhkgf0hcwg2&st=vw2afaq2&dl=0) every time and noted the number reported by dorado after basecalling has been completed.

dorado basecaller --device "cuda:0" sup pod5s/ >sup.bam

**Comment by Psy-Fer** [comment]
I wonder if that tinybox deployed as a server rack would trigger the "no datacentre deployment" clause in the ELUA

https://www.nvidia.com/content/DriverDownloads/licence.php?lang=us&type=GeForce

**Comment by ezherman** [comment]
To follow up: quick benchmark using @Kirk3gaard's data with a 5090 and dorado v0.9.5:

```
[2025-04-04 09:50:25.796] [info]  - downloading dna_r10.4.1_e8.2_400bps_sup@v5.0.0 with httplib
[2025-04-04 09:50:31.920] [info] > Creating basecall pipeline
[2025-04-04 09:50:32.762] [info] Calculating optimized batch size for GPU "NVIDIA GeForce RTX 5090" and model dna_r10.4.1_e8.2_400bps_sup@v5.0.0. Full benchmarking will run for this device, which may take some time.
[2025-04-04 09:50:34.957] [info] cuda:0 using chunk size 12288, batch size 224
[2025-04-04 09:50:35.099] [info] cuda:0 using chunk size 6144, batch size 224
[2025-04-04 10:06:13.875] [info] > Finished in (ms): 938619
[2025-04-04 10:06:13.875] [info] > Simplex reads basecalled: 180947
[2025-04-04 10:06:13.875] [info] > Simplex reads filtered: 85
[2025-04-04 10:06:13.875] [info] > Basecalled @ Samples/s: 1.439233e+07
[2025-04-04 10:06:13.950] [info] > Finished
```

**Comment by Kirk3gaard** [comment]
Cool. I  have added your data point here https://github.com/Kirk3gaard/2025-Crowdsource-GPU-basecalling-stats

Pretty decent performance from the 5090 😄 

![Image](https://github.com/user-attachments/assets/66947446-333d-497f-88cf-73ac6efa11ff)
